sound/soc/intel/boards/skl_hda_dsp_generic.c

```diff
@@ -135,11 +138,20 @@ static int skl_hda_fill_card_info(struct snd_soc_acpi_mach_params *mach_params)
 				skl_hda_be_dai_links[IDISP_DAI_COUNT +
 					HDAC_DAI_COUNT + i];
 		}
-	} else if (codec_count == 2 && codec_mask & IDISP_CODEC_MASK) {
+	} else if (codec_count <= 2) {
 		num_links = ARRAY_SIZE(skl_hda_be_dai_links);
 		num_route = ARRAY_SIZE(skl_hda_map);
 		card->dapm_widgets = skl_hda_widgets;
 		card->num_dapm_widgets = ARRAY_SIZE(skl_hda_widgets);
+		if (!(codec_mask & IDISP_CODEC_MASK)) {
+			if (codec_count == 2)
+				return -EINVAL;
+			for (i = 0; i < IDISP_DAI_COUNT; i++) {
+				skl_hda_be_dai_links[i].codecs = dummy_codec;
+				skl_hda_be_dai_links[i].num_codecs =
+					ARRAY_SIZE(dummy_codec);
```

Comment:
why not remove those HDMI dailinks, instead of mucking with the number of codecs?

Is this to avoid errors with the topology, where we are trying to fnd a dailink what would not exist if we removed them? And what happens if you just use dummy-codec, the user would still see an HDMI stream but the jack detection would never signal they can be used?

Reply:
@plbossart You guessed right. If we'd remove the links, then loading the topology fails. Requiring yet another topology for this case seems wasteful. 

With the dummy codec, "HDMI/DP Jack" ALSA controls are not created, so no jack detection events will be routed to user-space. The PCM devices will be created, but they will always fail at pcm open. And this kind of matches with what actually happens if ifgx is disabled... on SOF side we have the iDisp links, but the other end is just not connected.